app/server/lib/OIDCConfig.ts

```diff
@@ -167,33 +224,43 @@ export class OIDCConfig {
         profile,
       }));
 
-      delete mreq.session.oidc;
+      mreq.session.oidc = {
+        idToken: tokenSet.id_token, // keep idToken for logout
+      };
       res.redirect(targetUrl ?? '/');
     } catch (err) {
       log.error(`OIDC callback failed: ${err.stack}`);
-      // Delete the session data even if the login failed.
+      if (Object.prototype.hasOwnProperty.call(err, 'response')) {
+        log.error(`Response received: ${err.response?.body ?? err.response}`);
```

Comment:
According to https://github.com/panva/node-openid-client/blob/main/docs/README.md#errorresponse, `err.response` would be an `http.IncomingMessage`, which is really verbose and unhelpful to print out. How about, when no `.body`, falling back to `err.response.statusMessage` and/or `err.response.statusCode`?

Reply:
I realized that the error may be multiple things (especially the body may be a buffer).

I made some changes, with the hope it will be fine for you. If that's too tricky or you feel a bit uncomfortable, I may remove this log and reintroduce it a bit later or in a separate PR.